src/components/routerLink.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+  <a :href="resolve()" @click="onClick">
+    <slot />
+  </a>
+</template>
+
+<script setup lang="ts" generic="T extends string">
+  import { useRouter } from '@/compositions'
+  import { RouteMethod } from '@/types/routeMethod'
+  import { RegisteredRouteWithParams } from '@/types/routeWithParams'
+  import { RouterPushOptions } from '@/utilities/createRouterPush'
+
+  const props = defineProps<{
+    to: string | RegisteredRouteWithParams<T> | ReturnType<RouteMethod>,
+  } & RouterPushOptions>()
+
+  const router = useRouter()
+
+  function resolve(): string {
+    const thing = props.to
+
+    return router.resolve(thing)
```

Comment:
I'm assuming `thing` was for console logging

```suggestion
    return router.resolve(props.to)
```